src/operators/expression.h

```diff
@@ -85,31 +85,14 @@ class Expression {
   // Execute an arithmetic operator for two column of the chunk in the table
   template <typename ArrayType, typename ArrayPrimitiveType>
   arrow::Datum ExecuteBlock(int op, std::shared_ptr<arrow::Array> result,
-                            std::shared_ptr<arrow::Array> left_col,
-                            std::shared_ptr<arrow::Array> right_col);
+                            const std::shared_ptr<arrow::Array>& left_col,
+                            const std::shared_ptr<arrow::Array>& right_col);
 
   template <typename ArrayType, typename ArrayPrimitiveType>
-  arrow::Datum ExecuteBlock(bool is_result, const arrow::Scalar& scalar, int op,
+  arrow::Datum ExecuteBlockAPI(bool is_result, const arrow::Scalar& scalar, int op,
```

Comment:
I think we can remove the API suffix, since it is used only within the class to avoid confusion

Reply:
I was getting a compiler warning for the symbol not used on CLion. I changed it merely to surpress this warning. I can change this back.